lib/std/mem.zig

```diff
@@ -57,23 +86,64 @@ pub const Allocator = struct {
         /// Guaranteed to be a power of 2.
         /// Returned slice's pointer must have this alignment.
         new_alignment: u29,
-    ) Error![]u8,
+    ) Error![]u8 {
+        if (old_mem.len == 0)
+            return self.allocMem(new_byte_count, new_alignment);
+        if (new_byte_count == 0) {
+            self.freeMem(old_mem);
+            return old_mem[0..0];
+        }
+        if (isAligned(@ptrToInt(old_mem.ptr), new_alignment)) {
+            blk: {
+                self.resizeMem(old_mem, new_byte_count) catch |e| switch (e) {
+                    error.OutOfMemory => break :blk,
```

Comment:
Nitpick — this could be rewritten using if (error_union):
```zig
if (self.resizeMem(old_mem, new_byte_count)) |_| {
    return old_mem.ptr[0..new_byte_count];
} catch |_| {};
```